src/metrics/cognitive.rs

```diff
@@ -1696,4 +1733,197 @@ mod tests {
+    }
+
+    #[test]
+    fn java_compound_conditions() {
+        check_metrics::<JavaParser>(
+            "class X {
+              public static void print(boolean a, boolean b, boolean c, boolean d){  
+                if(a && b){ // +2
+                  System.out.println(\"test1\");
+                }
+                if(c && d){ // +2
+                  System.out.println(\"test2\");
+                }
+              }
+            }",
+            "foo.java",
+            |metric| {
+                insta::assert_json_snapshot!(
+                    metric.cognitive,
+                    @r###"
+                    {
+                      "sum": 4.0,
+                      "average": 4.0,
+                      "min": 0.0,
+                      "max": 4.0
+                    }"###
+                );
+            },
+        );
+    }
+
+    #[test]
+    fn java_switch_statement() {
+        check_metrics::<JavaParser>(
+            "class X {
+              public static void print(boolean a, boolean b, boolean c, boolean d){  // +1
+                switch(expr){ //+1
+                  case 1:
+                    System.out.println(\"test1\");
+                    break;
+                  case 2:
+                    System.out.println(\"test2\");
+                    break;
+                  default:
+                    System.out.println(\"test\");
+                }
+              }
+            }",
+            "foo.java",
+            |metric| {
+                insta::assert_json_snapshot!(
+                    metric.cognitive,
+                    @r###"
+                    {
+                      "sum": 1.0,
+                      "average": 1.0,
+                      "min": 0.0,
+                      "max": 1.0
+                    }"###
+                );
+            },
+        );
+    }
+
+    #[test]
+    fn java_switch_expression() {
+        check_metrics::<JavaParser>(
+            "class X {
+              public static void print(boolean a, boolean b, boolean c, boolean d){  // +1
+                switch(expr){ // +1
+                  case 1 -> System.out.println(\"test1\");
+                  case 2 -> System.out.println(\"test2\");
+                  default -> System.out.println(\"test\");
+                }
+              }
+            }",
+            "foo.java",
+            |metric| {
+                insta::assert_json_snapshot!(
+                    metric.cognitive,
+                    @r###"
+                    {
+                      "sum": 1.0,
+                      "average": 1.0,
+                      "min": 0.0,
+                      "max": 1.0
+                    }"###
+                );
+            },
+        );
+    }
+
+    #[test]
+    fn java_not_booleans() {
+        check_metrics::<JavaParser>(
+            "class X {
+              public static void print(boolean a, boolean b, boolean c, boolean d){  // +1
+                if (a && !(b && c)) { // +3 (+1 &&, +1 &&)
```

Comment:
We should change the comment otherwise the sum is not correct. As it is written right now, the sum is 2 not 3.

Reply:
Is there a reason for the change by the way? Both are valid. Change made though as I missed this one from previous request. Thanks.